### PR TITLE
Digital Ocean integration test for digital_ocean_tag

### PR DESCRIPTION
**Pull request by kbreit**
##### SUMMARY
None of the Digital Ocean modules have integration tests. This is an integration test for the tag module. There's very little around validation, but it at least catches obvious errors.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
digital_ocean_tag

##### ADDITIONAL INFORMATION
Credential information should go into the `integration_config.yml` file.

### CODE REVIEW

**Comment by kbreit** [comment]
@gundalow This is a duplicate PR for https://github.com/ansible/ansible/pull/49316. Should I use this one instead?

**Comment by gundalow** [comment]
@kbreit use whichever PR works best for you.

**Comment by kbreit** [comment]
@gundalow I'd like to use this one. Please review.

File: test/integration/targets/digital_ocean_tag/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  register: create_tag
+
+- debug:
+    var: create_tag
+
+- assert:
+    that:
+      - create_tag.changed == True
+      - create_tag.data.tag.name == "integration-test"
+
```
Comment (gundalow) [inline test/integration/targets/digital_ocean_tag/tasks/main.yml:21]: To confirm idempotent, create again and assert no changed.

Same for delete.

File: test/integration/targets/digital_ocean_tag/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    name: integration-test
+    state: absent
+  register: delete_tag
+
+- debug:
+    var: delete_tag
+
+- assert:
+    that:
+      - delete_tag.changed == True
```
Comment (gundalow) [inline test/integration/targets/digital_ocean_tag/tasks/main.yml]: Missing newline.

File: test/integration/targets/digital_ocean_tag/aliases [hunk 3]
```diff
@@ -0,0 +1,2 @@
+digital_ocean
+unsupported
```
Comment (gundalow) [inline test/integration/targets/digital_ocean_tag/aliases]: Missing newline.

File: test/integration/targets/digital_ocean_tag/tasks/main.yml (hunk 2 above)
Comment (kbreit) [reply test/integration/targets/digital_ocean_tag/tasks/main.yml]: Looks like the module has a idempotency bug so I can't make this integration test fully work. #51107

**Comment by kbreit** [comment]
@gundalow Please give this another look but I think we're ready to go on this one.

**Comment by BondAnthony** [comment]
LGTM

shipit

**Comment by kbreit** [comment]
bot_status

**Review by dagwieers — APPROVED** [review]
LGTM

File: test/integration/targets/digital_ocean_tag/tasks/main.yml (hunk 2 above)
Comment (kbreit) [reply test/integration/targets/digital_ocean_tag/tasks/main.yml]: Delete isn't idempotent.

File: test/integration/targets/digital_ocean_tag/aliases (hunk 3 above)
Comment (kbreit) [reply test/integration/targets/digital_ocean_tag/aliases]: Fixed

File: test/integration/targets/digital_ocean_tag/tasks/main.yml (hunk 1 above)
Comment (kbreit) [reply test/integration/targets/digital_ocean_tag/tasks/main.yml:21]: Added test for idempotency

File: test/integration/targets/digital_ocean_tag/tasks/main.yml (hunk 2 above)
Comment (dagwieers) [reply test/integration/targets/digital_ocean_tag/tasks/main.yml]: If delete is not idempotent and we can/will not fix this now, could you add a NOTE or a FIXME to the integration test to indicate this ?

```yaml
      # FIXME: Deleting a tag is not idempotent
      #- delete_tag_idempotent.changed == False
```

**Pull request merged**